### DISCUSSION TITLE
How to see inner exceptions, problems?

### DISCUSSION THREAD

**Discussion by andre-mitel** [thepirat000/Audit.NET#588] [Q&A]
I want to use the component with the MySQL data provider. Everything works fine.
But now I configure an invalid database to see what happens. Nothing happens, my code is working well without audit.
But my problem is I see no exceptions with a hint that the audit is not working and why. Thats not suitable for production.
Any ideas?

**Comment by thepirat000** [comment]
You can share a minimal sample project that reproduces the issue... The problem could be on your configuration

**Comment by andre-mitel** [comment]
Sure, here it is. Only a new webapp project with Audit.NET as configured in my project.
As you can see when I call AudioScope.LogAsync in row 36 Program.cs, I have no log entry that the audit log was not saved in the database because the database server is not available
[AuditNetTest.zip](https://github.com/thepirat000/Audit.NET/files/11048927/AuditNetTest.zip)
.

**Comment by thepirat000** [comment]
You are not awaiting the LogAsync call, that's why you miss the exception

**Reply by andre-mitel** [reply]
Yes correct. I need the audit events processed async from my runtime code. Now I make my own async approach and use the AuditScope.Log synchron method and catch and log the exception myself. Thanks
